### PR TITLE
[Console] Improve markdown format

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | "master"
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | not sure?
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any
| License       | MIT
| Doc PR        | reference to the documentation PR, if any

This improves the markdown description for a console application. To make the ouput read more friendly and intuitively (less bloated IMHO).

Before:

Markdown files in https://github.com/symfony/symfony/tree/master/src/Symfony/Component/Console/Tests/Fixtures

After:

Markdown files in https://github.com/ro0NL/symfony/tree/console/markdown/src/Symfony/Component/Console/Tests/Fixtures

### CODE REVIEW

File: src/Symfony/Component/Console/Tests/Fixtures/application_1.md [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+============
 
-* help
-* list
+* `help`
+* `list`
 
-help
-----
+`help`
```
Comment (ogizanagi) [inline src/Symfony/Component/Console/Tests/Fixtures/application_1.md]: This should rather be a second level title IMHO.

**Comment by ogizanagi** [comment]
To me it isn't a BC break, because the markdown format is not something meant to be analyzed programmatically and only intends to provide a quick documentation of the console application (I'm not even sure it'll be considered as a BC break for json actually, nor any format produced by a command is part of the BC promise).

Status: Reviewed 👍

**Comment by ogizanagi** [comment]
Status: Reviewed

**Comment by javiereguiluz** [comment]
@ogizanagi I agree that this should't be considered a BC. By the way, a while ago I thought about opening a RFC issue to discuss about removing this feature. Do people really use this Markdown descriptor?

**Comment by ro0NL** [comment]
It's nice API doc for a command. Ie. i would use this to generate a README.md for a command repo.

**Comment by ogizanagi** [comment]
I think it's an easy way to provide a minimalistic documentation for a tool and it worth keeping it.

I don't see as much utility for the json/xml descriptors though. I don't know what usages it deserves. Someone may try to implement a UI by parsing programmatically those formats, or maybe it can be consumed by generic worker in order to know which command and how to call, but I never seen someone doing nor asking for this.

**Comment by fabpot** [comment]
Guys, before discussing removing something, you should assume that it was added for a reason :) The XML format is used by tools like IDEs to get information about available commands to provide a nice UI for instance.

**Comment by javiereguiluz** [comment]
Yes, the XML export is useful for that cases. But what about the Markdown export?

**Comment by fabpot** [comment]
IIRC, the Markdown format has been added quite recently. So, let's dig a bit into the past PR to see why it was added.

**Comment by ogizanagi** [comment]
All formats were added in the same PR by @jfsimon: https://github.com/symfony/symfony/pull/7887

**Comment by ro0NL** [comment]
This PR only tends to improve what we already have :) (application_2.md goes from 376 to 350 lines btw).

**Comment by ro0NL** [comment]
What about removing `Is value required` if an option accepts no value?

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Comment by javiereguiluz** [comment]
I was late to this PR, but I was just about to propose the following improvements.

This is how an app description looks like:

before

And here are some comments:

after

**Comment by fabpot** [comment]
If the commands are hidden, I think they should not be part of the Mardown format but still be listed in the JSON or XML formats. The idea is that end users should not see them but machines should. 👍 for your other suggestions.

**Comment by ro0NL** [comment]
I believe hidden commands are already skipped indeed.

The application is actually included now in the markdown title ([ref](https://github.com/symfony/symfony/pull/20866/files#diff-cbff8bda6ae373aa5d4fedf069ee9e34R126)), it wasnt before. However the default version (`UNKNOWN`) is not shown.

I think we're good :)

**Pull request merged**